substrate/client/network/sync/src/strategy/state_sync.rs

```diff
@@ -132,6 +132,78 @@ where
 			skip_proof,
 		}
 	}
+
+	fn process_state_key_values(
+		&mut self,
+		state_root: Vec<u8>,
+		key_values: impl IntoIterator<Item = (Vec<u8>, Vec<u8>)>,
+	) {
+		let is_top = state_root.is_empty();
+
+		let (child_key_values, top_key_values): (Vec<_>, Vec<_>) =
+			key_values.into_iter().partition(|key_value| {
+				is_top && well_known_keys::is_child_storage_key(key_value.0.as_slice())
```

Comment:
```suggestion
				!is_top && well_known_keys::is_child_storage_key(key_value.0.as_slice())
```

Reply:
Hmm, are you sure about this? The original logic is `is_top`.

Reply:
Ahh yeah you are right, this code is a little bit confusing :see_no_evil: